validator/vars.go

```diff
@@ -132,11 +133,12 @@ func (v *varValidator) validateVarType(typ *ast.Type, val reflect.Value) (reflec
 		kind := val.Type().Kind()
 		switch typ.NamedType {
 		case "Int":
-			if kind == reflect.String || kind == reflect.Int || kind == reflect.Int32 || kind == reflect.Int64 {
+			_,e := strconv.ParseInt(fmt.Sprintf("%v", val.Interface()), 10, 64);
+			if e == nil && (kind == reflect.Int || kind == reflect.Int32 || kind == reflect.Int64 || kind == reflect.Float32 || kind == reflect.Float64){
 				return val, nil
 			}
 		case "Float":
-			if kind == reflect.String || kind == reflect.Float32 || kind == reflect.Float64 || kind == reflect.Int || kind == reflect.Int32 || kind == reflect.Int64 {
+			if kind == reflect.Float32 || kind == reflect.Float64 || kind == reflect.Int || kind == reflect.Int32 || kind == reflect.Int64 {
```

Comment:
I am not sure why you removed the `kind == reflect.String` condition as "45.230343" strings are often how they are serialized.

Reply:
the tests return an error because you are giving a string to an Int.

I removed the string from the string because if you set a variable as int or float you should get an int or a float, not a string, in fact, I got errors for that too. I hope it is useful for something.

Reply:
still, adding the string without removing the validation on the Int, there is no problem. the bad thing is that it can return a string in an int or float.